### DISCUSSION TITLE
Is it possible to make a PathLayer with separate segments from binary data?

### DISCUSSION THREAD

**Discussion by jwhitaker-swiftnav** [visgl/deck.gl#7390] [Q&A]
It seems impossible, unless I'm missing something:
```js
const data = {
  length: 16,
  startIndices: [0, 4, 8, 12],
  attributes: {
    getPath: {
      value: new Float32Array([
        -122.00,37.00, -122.01,37.01, -122.02,37.02, -122.03,37.03,
        -121.97,37.03, -121.98,37.02, -121.99,37.01, -122.00,37.00,
        -122.00,37.00, -121.99,36.99, -121.98,36.98, -121.97,36.97,
        -122.02,36.97, -122.02,36.98, -122.01,36.99, -122.00,37.00,
      ]),
      size: 2
    },
    getColor: {
      value: new Uint8ClampedArray([
        255,255,0,255, 255,255,0,255, 255,255,0,255, 255,255,0,255,
        255,0,255,255, 255,0,255,255, 255,0,255,255, 255,0,255,255,
        0,255,255,255, 0,255,255,255, 0,255,255,255, 0,255,255,255,
        128,128,255,255, 128,128,255,255, 128,128,255,255, 128,128,255,255
      ]),
      size:4
    }
  }
}

const layer = new PathLayer({
  id: 'PathLayer',
  data: data,
  positionFormat: 'XY',
  _pathType: "open"
  ...
}
```

Result: 
![image](https://user-images.githubusercontent.com/113399675/200497515-40418941-14da-45d1-9981-5f354ae0449a.png)

Full codepen:  https://codepen.io/akdor1154/pen/wvXoyYZ?

I would expect the result should have four visible lines, of Yellow, Purple, Teal, and Light Blue, and each of the lines should not be joined with the others (though they coincidently do share the centre point). Is this a bug or am I holding it wrong?

EDIT: I'm holding it wrong, startIndices off by one.

**Comment by linze99** [comment]
Did you mean this?
![line](https://user-images.githubusercontent.com/30024479/200504281-fe975022-3403-47b5-998d-eb88a70e1100.PNG)
you can use LineLayer instead .
Full codepen: [https://codepen.io/linze/pen/gOKLeeO](url)
